code/script.js

```diff
@@ -1,43 +1,214 @@
 
-// If you need any global variables that you can use across different functions, declare them here:
-
-
+//Global variables. I have one for userName to be able to use the value in more than first reply.
+let userName;
 // Declare your functions after this comment
 
 // This function will add a chat bubble in the correct place based on who the sender is
+//without the ${message} the actual message(parameter) in const greetUser will not be passed, only string message. Only the parameter 'bot' 
 const showMessage = (message, sender) => {
-  // the if statement checks if the sender is 'user' and if that's the case it inserts an html senction inside the chat with the posted message
+  // the if statement checks if the sender is 'user' and if that's the case it inserts an html section inside the chat with the posted message
   if (sender === 'user') {
+    console.log("Now the user is replying");
     chat.innerHTML += `
       <section class="user-msg">
         <div class="bubble user-bubble">
           <p>${message}</p>
         </div>
-        <img src="assets/user.png" alt="User" />  
+        <img src="assets/user-image.png" alt="User" />  
       </section>
-    `
-    // the else if statement checks if the sender is a bot and if that's the case it inserts an html senction inside the chat with the posted message
+    `;
+    // the else if statement checks if the sender is a bot and if that's the case it inserts an html section inside the chat with the posted message
   } else if (sender === 'bot') {
+    console.log("Now the bot is asking a question");
     chat.innerHTML += `
       <section class="bot-msg">
-        <img src="assets/bot.png" alt="Bot" />
+        <img src="assets/bot-image.png" alt="Bot" />
         <div class="bubble bot-bubble">
           <p>${message}</p>
         </div>
       </section>
-    `
+    `;
   }
   // This little thing makes the chat scroll to the last message when there are too many to be shown in the chat box
   chat.scrollTop = chat.scrollHeight;
 }
 
-// Starts here
+// My "Conversation" for the Chat starts here
 const greetUser = () => {
-  // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot');
-  // Just to check it out, change 'bot' to 'user' here 👆
+  // here we call the function showMessage, that we declared earlier. With the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
+  showMessage("Welcome, What's your name?", 'bot');
+}
+
+// Initial button click, here I should get the name entered
+sendBtn.addEventListener('click', (event) => {
+  if (nameInput.value === "" || undefined) {
+    showMessage("Sorry I did not catch that, please enter your name!", 'bot')
+    event.preventDefault()
+  } else {
+    event.preventDefault() //this is so the page do not reload.
```

Comment:
In this event listener you have `event.preventDefault()`  in both blocks of code; maybe move it outside the if...else block? Then you wont need the else statement
Something like this:
```
sendBtn.addEventListener('click', (event) => {
  event.preventDefault()
  if (nameInput.value === "" || undefined) {
   ....
  }
```

Reply:
Thank you for suggestion, I have now moved it to be outside of the if else block. I think I still need to keep the else statement, there is what happens if the name is entered. It might be a bit hard to see with all the comments

```sendBtn.addEventListener('click', (event) => {
  event.preventDefault() 
  if (nameInput.value === "" || undefined) {
    showMessage("Sorry I did not catch that, please enter your name!", 'bot')
    event.preventDefault()
  } else {
    // Store the value in a variable so I can access it after we clear it from the input. 
    // The variable were the value will be stored is outside of this function to be able to use it in more places. In here it get assigned
    userName = nameInput.value
    showMessage(`${userName}`, 'user');
    // Clears the input field
    nameInput.value = ''
    //Here I call the function where I present the dishes to choose from. I will also pass the userName
    setTimeout(() => whatKindOfDish(userName), 1000)
  }
})```